torch/fx/node.py

```diff
@@ -648,9 +645,18 @@ def is_impure(self):
         if self.op in {"placeholder", "output"}:
             return True
 
-        # Check if an impure function.
+        # Check if an impure function based on schema.
+        # A call_function is impure if it has at least one mutable argument that has more
+        # than 1 users (all arguments have `self` as a user).
         if self.op == "call_function":
-            return self.target in _side_effectful_functions
+            schema = getattr(self.target, "_schema", None)
+            schema_mutable = False
+            if schema is not None and schema.is_mutable:
+                for (idx, arg) in enumerate(schema.arguments):
+                    if arg.alias_info is not None and arg.alias_info.is_write:  # arg is mutable
+                        if len(self.args[idx].users) > 1:  # has other users except self
+                            schema_mutable = True
```

Comment:
what about kwargs?

Reply:
added kwargs now.

Reply:
oh wait, I think there's some issues with the kwargs implementation. I'll fix it and let you know.

Reply:
should be fixed now!